score/liquid_icx/tests/test_integrate_liquid_icx.py

```diff
@@ -30,71 +30,111 @@ def test_score_update(self):
             self.assertEqual(self._score_address, tx_result['scoreAddress'], msg=pp.pformat(tx_result))
 
     def test_0_join_delegate_stake_fallback(self):
+        """
+        1. Join as owner and then join with another 10 random generated wallets and check the length of the linkedlist
+        2. Perform detailed owner's wallet check
+        3. Check staked, delegate values
+        4. Try to send ICX directly to contract ( fallback function )
+        """
+        # 1
         self.assertEqual(self._get_wallets(), [])
         self._join()
         self.assertEqual(len(self._get_wallets()), 1)
         self._n_join(10)
         self.assertEqual(len(self._get_wallets()), 11)
         self._join()
         self.assertEqual(len(self._get_wallets()), 11)
+        # 2
         owner = self._get_wallet()
         self.assertEqual(owner["locked"], hex(2 * 10 * 10**18), msg=pp.pformat(owner))
         self.assertEqual(len(owner["join_values"]), 2, msg=pp.pformat(owner))
         self.assertEqual(len(owner["unlock_heights"]), 2, msg=pp.pformat(owner))
+        self.assertEqual(len(owner["join_values"]), 2, msg=pp.pformat(owner))
+        # 3
         self.assertEqual(self._get_staked(), hex(12 * 10 * 10**18), msg=pp.pformat(owner))
         self.assertEqual(self._get_delegation()["totalDelegated"], hex(12 * 10 * 10**18), msg=pp.pformat(owner))
-        owner = self._get_wallet()
-        self.assertEqual(len(owner["join_values"]), 2, msg=pp.pformat(owner))
+        # 4
         fallback_tx = self._transfer_icx_from_to(self._wallet, self._score_address, value=5, condition=False)
         self.assertIn("LICX does not accept ICX", fallback_tx["failure"]["message"])
```

Comment:
imo it's easier to find bugs and understand the tests if one test case only tests a single functionality.
E.g. I would make an own test case for `fallback`, or in `test_1_join_balance_transfer_leave` I wouldn't assert the `join` TX as this would already be done via `test_0_join_delegate_stake_fallback`.